packages/devtools_app/lib/src/ui/flutter/service_extension_widgets.dart

```diff
@@ -182,14 +184,43 @@ class HotRestartButton extends StatelessWidget {
   Widget build(BuildContext context) {
     return _RegisteredServiceExtensionButton._(
       serviceDescription: hotRestart,
-      action: serviceManager.performHotRestart,
+      action: () =>
+          _wrapReloadCall('restart', serviceManager.performHotRestart),
       inProgressText: 'Performing hot restart',
       completedText: 'Hot restart completed.',
       describeError: (error) => 'Unable to hot restart the app: $error',
     );
   }
 }
 
+Future<void> _wrapReloadCall(
+  String name,
+  Future<void> Function() reloadCall,
+) async {
+  try {
+    final Stopwatch timer = Stopwatch()..start();
+    messageBus.addEvent(BusEvent('$name.start'));
+    await reloadCall();
+    timer.stop();
+    // 'restarted in 1.6s'
+    final String message = '${name}ed in ${_renderDuration(timer.elapsed)}';
+    messageBus.addEvent(BusEvent('$name.end', data: message));
+    // TODO(devoncarew): Add analytics.
+    //ga.select(ga.devToolsMain, ga.hotRestart, timer.elapsed.inMilliseconds);
+  } catch (_) {
+    final String message = 'error performing $name';
+    messageBus.addEvent(BusEvent('$name.end', data: message));
+  }
+}
+
+String _renderDuration(Duration duration) {
```

Comment:
move this to src/utils.dart with similar helpers for rendering durations.
Kenzie, should we update the existing `msText` method in that util class to also be smart about rendering time intervals larger than 1 second or for the timeline is there a good reason to stick with ms even for very large numbers of ms.

Reply:
It may look a little odd in the Timeline where we display the unit after the number value if we change from `ms` to `s`
<img width="253" alt="Screen Shot 2020-03-17 at 12 34 11 PM" src="https://user-images.githubusercontent.com/43759233/76894566-9fbcae00-684b-11ea-9e75-26d53ea56e1d.png">

but we could always add a boolean to the method that does this and default to false

Reply:
Moved to `utils.dart`.